Update Array.p[@@unscopables] list (#19697)
mdn/content: files/en-us/web/javascript/reference/global_objects/array/@@unscopables/index.md

The default `Array` properties that are ignored for `with` statement-binding purposes are:

- [`at()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at)
- [`copyWithin()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/copyWithin)
- [`entries()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/entries)
- [`fill()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/fill)
- [`find()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find)
- [`findIndex()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex)
- [`findLast()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findLast)
- [`findLastIndex()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findLastIndex)
- [`flat()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat)
- [`flatMap()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap)
- [`includes()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes)
- [`keys()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/keys)
- [`values()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/values)

> `Array.prototype[@@unscopables]` is an object with `null` prototype, where all the above property names have value `true`.

Reviewer: I like this sentence! (seems like so far it wasn't spelled out what is actually returned here!)

Not sure if we need to say that it's a null prototype. Simpler alternative to consider:

```suggestion
`Array.prototype[@@unscopables]` is an object containing all the above property names with the value `true`.
```

Author: Thanks—I don't have strong opinions about mentioning the prototype or not (although it does matter, otherwise `toString` would be unscopable as well).

Reviewer: Ah yes, that makes a lot of sense. Well, with that in mind, how about spelling it out like this?

```suggestion
`Array.prototype[@@unscopables]` is an empty object only containing all the above property names with the value `true`.
```

> See {{jsxref("Symbol.unscopables")}} for how to set unscopable properties for your own objects.

Reviewer: Another thing to mention could be that built-in DOM APIs can set this sometimes, too. For example: https://developer.mozilla.org/en-US/docs/Web/API/Element/remove

Author: That could be mentioned on the `Symbol.unscopables` page itself?

Reviewer: Agreed, probably the better page for that.

Author: There are a few other well-known symbols used by DOM APIs (like `@@isConcatSpreadable` used by `NodeList`) that's worth mentioning. I think I'll do that some time later
